### PR TITLE
Removing id generation from setter (and generic setter)

### PR DESCRIPTION
**Pull request by brikou**

### CODE REVIEW

File: src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig [hunk 1]
```diff
@@ -10,9 +10,11 @@
     public function setByName($name, $value)
     {
 {% for fieldMapping in metadata.fieldMappings %}
+{% if fieldMapping.id is not defined %}
```
Comment (fzaninotto) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: Why?

Comment (brikou) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: because id is auto generated, am i wrong?

Comment (fzaninotto) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: id means primary key, not necessarily auto-generated. BEsides, the primary key can be composite.

Comment (brikou) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: ok but how can we check that this key is (auto generated + not composite) (most of the time i guess it is autogenerated and not composite but simple case)

Comment (fzaninotto) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: I think that's irrelevant. Even doctrine's [EntityGenerator](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/Tools/EntityGenerator.php) doesn't do that.

Comment (brikou) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: But if it was possible would you find it great ?

Comment (fzaninotto) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: No, even autogenerated Pks can be manually set sometimes (at least in MySQL), so removing the setter removes a feature that is useful.

Comment (brikou) [inline src/Propel/Builder/ORM/templates/BaseActiveRecord/GenericGetterSetter.php.twig:13]: ok i get it

2011/5/4 fzaninotto <
reply@reply.github.com>

> > @@ -10,9 +10,11 @@
> >      public function setByName($name, $value)
> >      {
> >  {% for fieldMapping in metadata.fieldMappings %}
> > +{% if fieldMapping.id is not defined %}
> 
> No, even autogenerated Pks can be manually set sometimes (at least in
> MySQL), so removing the setter removes a feature that is useful.
> 
> ## 
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/fzaninotto/Propel2/pull/10/files#r25718

## 
- Brice -

**Pull request closed without merging**